proto/keystore_api/v1/keystore.proto

```diff
@@ -0,0 +1,150 @@
+}
+
+// A light pointer for a conversation that contains no decryption keys
+message ConversationReference {
+    string topic = 1;
+    uint64 created_ns = 2;
+    xmtp.message_contents.InvitationV1.Context context = 3;
+}
+
+// Decrypt a batch of messages using X3DH key agreement
+message DecryptV1Request {
+    // A single decryption request
+    message Request {
+        xmtp.message_contents.Ciphertext payload = 1;
+        xmtp.message_contents.PublicKeyBundle peer_keys = 2;
+        bytes header_bytes = 3;
+        bool is_sender = 4;
+    }
+
+    repeated Request requests = 1;
+}
+
+// Response type for both V1 and V2 decryption requests
+message DecryptResponse {
+    // A single decryption response
+    message Response {
+        // Wrapper object for success response
+        message Success {
+            bytes decrypted = 1;
+        }
+
+        oneof response {
+            Success result = 1;
+            KeystoreError error = 2;
+        }
+    }
+
+    repeated Response responses = 1;
+}
+
+// Decrypt a batch of messages using the appropriate topic keys
+message DecryptV2Request {
+    // A single decryption request
+    message Request {
+        xmtp.message_contents.Ciphertext payload = 1;
+        bytes header_bytes = 2;
+        string content_topic = 3;
+    }
+
+    repeated Request requests = 1;
+}
+
+// Encrypt a batch of messages using X3DH key agreement
+message EncryptV1Request {
+    // A single encryption request
+    message Request {
+        xmtp.message_contents.PublicKeyBundle recipient = 1;
+        bytes payload = 2;
+        bytes header_bytes = 3;
+    }
+
+    repeated Request requests = 1;
+}
+
+// Response type for both V1 and V2 encryption requests
+message EncryptResponse {
+    // A single encryption response
+    message Response {
+        // Wrapper object for success response
+        message Success {
+            xmtp.message_contents.Ciphertext encrypted = 1;
+        }
+
+        oneof response {
+            Success result = 1;
+            KeystoreError error = 2;
+        }
+    }
+
+    repeated Response responses = 1;
+}
+
+// Encrypt a batch of messages using the appropriate topic keys
+message EncryptV2Request {
+    // A single encryption request
+    message Request {
+        bytes payload = 1;
+        bytes header_bytes = 2;
+        string content_topic = 3;
+    }
+
+    repeated Request requests = 1;
+}
+
+// Request to create an invite payload, and store the topic keys in the Keystore
+message CreateInviteRequest {
+    xmtp.message_contents.InvitationV1.Context context = 1;
+    xmtp.message_contents.SignedPublicKeyBundle recipient = 2;
+    uint64 created_ns = 3;
+}
+
+// Response to a CreateInviteRequest
+message CreateInviteResponse {
+    ConversationReference conversation = 1;
+    bytes payload = 2;
+}
+
+// Request to save a batch of invite messages to the Keystore
+message SaveInvitesRequest {
+    repeated xmtp.message_api.v1.Envelope envelopes = 1;
```

Comment:
Echoing my outdated comment I made moments ago: https://github.com/xmtp/xmtp-js/pull/267/files#r1099023226

Can we exclude message_api protos from the Keystore API protos? It makes proto generation have more dependencies for Rust and doesn't add anything AFAICT. It'll look ugly to just have repeated byte arrays as this field, but I think it would work.

Reply:
I'm alright with removing the dependency on `message_api`, since it's just used in one place. I wouldn't want to remove all the `message_contents` references though. We have a lot of those, and I think it's useful to be able to have consistent types across the SDK wherever possible. Re-implementing those feels like it would create a bunch of boilerplate, and new places for bugs to creep in.

Reply:
If our only concern is the GRPC dependency, happy to remove this and create a new type that just has the topic and the message.

Reply:
No changes necessary then! If it's not an obvious move then I don't think it's worth cutting out message_api dependency

Reply:
Oh wait sorry I misread your comment. Yeah we definitely don't need to replace `message_contents` references. Those feel natural and right to build upon.